---
title: "MSB7102 Mini-project, Semester I, 2021"
author: "kikomeko Aloysious"
date: "6/23/2021"
output:
  pdf_document: default
  word_document: default
  html_document: default
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


<!-- Background -->
<!-- This is an assessed exercise that will contribute to coursework for this module. It is aimed at providing an experience with all that has been covered throughout the semester. The tasks are based on two Bioconductor packages; phyloseq and DEseq2. Endeavor to look at the documentation and the links indicated below may be useful. The data files are located in the shared google drive folder. -->
<!-- Data sources and description -->
<!-- The data used in this exercise is derived and was generated by Kolistic et al in their study “The dynamics of the human infant gut microbiome in development and progression toward Type 1 Diabetes”. See full publication at https://doi.org/10.1016/j.chom.2015.01.001. Briefly, this was a prospective analysis of developing gut microbiome in infants en route to type 1 diabetes. Infants from Finland and Estonia were recruited at birth based on predisposition to autoimmunity determined by human leukocyte antigen (HLA) genotyping. The cohort consists of 33 infants, 11 of which seroconverted to serum autoantibody positivity and of those, four developed T1D within the three-year time-frame of this study. -->
<!-- Tasks  -->
<!--     1. Import the data described above into R, provide descriptive summaries of the subject data (using appropriate graphics and statistical summary measures) given in the diabimmune_16s_t1d_metadata.csv file. In addition, use appropriate test(s) to check for association/independency between disease status and other variables (delivery mode, gender and age). Note that age is given in days. -->
<!--     2. Using phyloseq, create a phyloseq object. This will comprise the OTU abundance, taxonomy (provided in the .txt file) and sample data (provided in the .csv file). -->
<!--     3. Generate Alpha diversity plots and ordination plots. Examine any observed patterns by delivery mode, gender and disease status. -->
<!--     4. Perform differential abundance using DEseq2  -->
<!-- Useful links -->
<!--     • Importing data: https://joey711.github.io/phyloseq/import-data.html  -->
<!--     • Ordination: https://joey711.github.io/phyloseq/plot_ordination-examples.html  -->
<!--     • Alpha diversity: https://joey711.github.io/phyloseq/plot_richness-examples.html  -->
<!--     • Differential abundance: http://joey711.github.io/phyloseq-extensions/DESeq2.html  -->
<!-- Submission -->
<!-- Use Rmarkdown to generate a report for your submission. Generate the report as a pdf or html file, upload it to your GitHub repository, send the link to submissions.gronald@gmail.com and copy assekagiri@gmail.com by 25th-06-2021 under the subject “MSB7102 Mini-project, semester I, 2021”. -->




------

![](/home/biompigi/Documents/bioinformstics2020/R/R_CW-25-06-2021/Mini-project/Makerere_emblame.png)

<div align="center">**COLLEGE OF HEALTH SCIENCES**</div>

<div align="center">**SCHOOL OF BIOMEDICAL SCIENCES**</div>

<div align="center">**DEPARTMENT OF IMMUNOLOGY AND MOLECULAR BIOLOGY**</div>

<div align="center">**MASTER OF SCIENCE IN BIOINFORMATICS**</div>

<div align="center">**COURSE UNIT: Bioconductor & R for Boioinformatics**</div>
<div align="center">**COURSE CODE: MSB 7102**</div>

<div align="center">**STUDENT NAME: Kikomeko Aloysious**</div>
<div align="center">**Reg. No. 2019/HD07/24856U**</div>  

<div align="center">**Student No. 1900724856**</div>

<div align="center">**EMAIL: kikomekoa2013@gmail.com**</div>

----------------------------------------------------------------------------------------------------------
<!-- Tasks -->  

```{r include=FALSE, parkages, echo=FALSE}

library(tidyverse)
library(table1)
library(readxl)
library(phyloseq)
library(DESeq2)
library(curl)
library(later)
library(ggplot2)
library(rgl)
library(barplot3d)
library(readr)
library(readxl)
library(rlang)
```


-------------------------------------------------------------------------------------------------------
**<div align="left">1. Import the data described above into R, **


```{r t1d-metedata}

t1d_metadata <- read.csv("diabimmune_16s_t1d_metadata.csv",
              stringsAsFactors = FALSE)

```

**Change structure of categorical variables to factors**

```{r data-structure}
t1d_metadata$Gender <- as.factor(t1d_metadata$Gender)  
t1d_metadata$Case_Control <- as.factor(t1d_metadata$Case_Control)
t1d_metadata$Delivery_Route <- as.factor(t1d_metadata$Delivery_Route)
```


# **i.    provide descriptive summaries of the subject data (using appropriate graphics and statistical summary measures) given in the diabimmune_16s_t1d_metadata.csv file.**

```{r t1d_metadata}
summary(t1d_metadata)
```


**Cross tabulation Gender Vs Age at enrollment**

```{r crosstabulation}
addmargins(xtabs(~Gender + Case_Control, data = t1d_metadata))
```

The table above show that, more females (412) were involved in the study than the males (365). A total of 260 respondents (cases), 142 were female and 118 were male. Out of the 517 respondents (Controls), 270 were female and 365 were male 



```{r crosstabulation3}
addmargins(xtabs(~Delivery_Route+Case_Control, data = t1d_metadata))
```
from the table above, its noted that no baby born by Cesarian section was included in the sample as a case, but 66 babies born by cesarian section were sampled as control elements. 260 babies were  


## **Graphics**


```{r}
plot(t1d_metadata$Case_Control, t1d_metadata$Age_at_Collection, main="Disease Diversity and Age ",
    xlab="Case_Control ", ylab="Age_at_Collection ", pch=19)
 
```
From the box plot above, it can be seen that, the age at collection for both cases and control groups, were almost at the same level of the lower quartile, Although, the study elements under cases had a higer median value and a higher upper quartile value but with a lower value in the upper and higher value for the lower whiskers. unlike in the control groups somewhat a lower median value, and lower upper quartile, they had a higher value in the upper whisker lower value in the lower whisker.




```{r bar_graph}
library (ggplot2)
bar_chart1 <- ggplot(t1d_metadata, aes(x =Case_Control ,y= ((..count..)/sum(..count..)), fill= Gender)) + 
  geom_bar(position = "dodge", color= "Black")+
  scale_y_continuous(labels=scales::percent)+ 
  labs(title = "Percentage of Disease Status by Gender",
       y= "Percentage", x= "Disease Status" )+
  geom_text(stat= "count", aes(label= scales::percent((..count..)/sum(..count..))), position = position_dodge(0.9), size=4, vjust=-0.1)+
  theme_classic()+
  scale_fill_brewer(palette = "PiYG")
  bar_chart1
```
From the graph above, among the cases, 18.3% were female and 15.2% were male. Among the study control elements, 34.7% were female and 31.8% were male.

```{r }
bar_chart2 <- ggplot(t1d_metadata, aes(x =Case_Control ,y= (..count..)/sum(..count..), fill= Delivery_Route)) + 
  geom_bar(position = "dodge", color= "Black")+
  scale_y_continuous(labels=scales::percent)+ 
  labs(title = "Percentage of Disease Status by Delivery Route",
       y= "Percentage", x= "Disease Status" )+
  geom_text(stat= "count", aes(label= scales::percent((..count..)/sum(..count..))), position = position_dodge(0.9), size=4, vjust=-0.1)+
  theme_classic()+
  scale_fill_manual(name = "", values = c("blue", "red"))
  bar_chart2
```
The graph above shows that, 33% of the study population were cases and had been produced through vaginal delivery, 8% of the study population among controls, had been produced through cesarian section, and 58% among control study elements had been produced through the vagianal route. 
```{r}
bar_chart3 <- ggplot(t1d_metadata, aes(x =Delivery_Route ,y= (..count..)/sum(..count..), fill= Age_at_Collection)) + 
  geom_bar(position = "dodge",color= "red")+
  scale_y_continuous(labels=scales::percent)+ 
  labs(title = "Percentage of Participants by Age and Disease_Status",
       y= "Percentage", x= "Delivery Route")+
  geom_text(stat= "count", aes(label= scales::percent((..count..)/sum(..count..))), position = position_dodge(0.9), size=4, vjust=-0.1)+
  theme_classic()+
  scale_fill_manual(name = "", values = ("black")) 
  bar_chart3
```

Its evident that, only 8% of the study population were born by cesarian section and 92% were born by vaginal delivery.



ii. In addition, use appropriate test(s) to check for association/independency between disease status and other variables (delivery mode, gender and age). Note that age is given in days.

```{r Tests-of-Independence}
df1 <- xtabs(~Gender + Case_Control, data = t1d_metadata)
 
chisq.test(df1)
fisher.test(df1)
```
The Chi-squared probability value of 0.5796, like the Fisher's Exact Test p-value (0.543), which is greater than 5% level of significance,  is insignificant at the 95% level of significance, implying that gender and case_Control were independent variables.

```{r Tests-of-Independence2}
df2 <- xtabs(~Delivery_Route + Case_Control, data = t1d_metadata)
 
chisq.test(df2)
fisher.test(df2)

```


The Chi-squared probability value of 3.949e-09, like the Fisher's Exact Test p-value (6.86e-13), which is less than 5% level of significance, is significant at the 95% level of significance, implying that Delivery_route and case_Control were dependent variables.

```{r Tests-of-Independence3}
df3 <- xtabs(~Age_at_Collection+ Delivery_Route , data = t1d_metadata)
 
t.test(df3)
```
The Chi-squared probability value (0.5186), like the Fisher's Exact Test p-value (0.5195), which is greater than 5% level of significance, is insignificant at the 95% level of significance, implying that gender and case_Control were independent variables.


```{r}
bar_chart4 <- ggplot(t1d_metadata, aes(x =Case_Control ,y= (..count..)/sum(..count..), fill= Age_at_Collection)) + 
  geom_bar(position = "dodge",color= "yellow")+
  scale_y_continuous(labels=scales::percent)+ 
  labs(title = "Percentage of Participants by case_control and Age",
       y= "Percentage", x= "case_Control")+
  geom_text(stat= "count", aes(label= scales::percent((..count..)/sum(..count..))), position = position_dodge(0.9), size=4, vjust=-0.1)+
  theme_classic()+
  scale_fill_manual(name = "", values = c("black")) 
  bar_chart4
```

Its evident that, the biggest proportion of the respondents 67%, were considered as controls irrespective of the age, while 33% of the study population were cases in the study.


```{r}
model = glm(t1d_metadata$Case_Control ~ t1d_metadata$Gender + t1d_metadata$Delivery_Route + t1d_metadata$Age_at_Collection,  family=binomial)
summary(model)
anova(model, test = "Chisq")
x <- t1d_metadata$Age_at_Collection
y <- dnorm(t1d_metadata$Age_at_Collection, mean(t1d_metadata$Age_at_Collection), sd(t1d_metadata$Age_at_Collection))
plot(x, y)

plot(t1d_metadata$Delivery_Route, t1d_metadata$Case_Control)

```
From the linear regression model above, we can conclude that, inclusion the study elements was significantly depended on the deliverly route unlike the Age_at_enrollment and gender.

The normal distribution curve indicates that, the age of most of the study elements were 600 days of life old and bellow. This is due to the fact that, most of the data is left skewed.

The stacked bar plot shows that, all study elements delivered by cesarian section were considered as controls, and only those produced through spontaneous vaginal delivered, were used as both cases and controls.

```{r}
model1 = anova(model, test = "Chisq")
summary(model1)
```

```{r}
model1 = anova(model, test = "Cp")
summary(model1)
```

# **2. Using phyloseq, create a phyloseq object. This will comprise the OTU abundance, taxonomy (provided in the .txt file) and sample data (provided in the .csv file).**


```{r}
library(tidyverse)
taxa.data <- read_tsv("diabimmune_t1d_16s_otu_table.txt", skip = 1)
otu.data <- taxa.data %>% select(1:778)
taxonomy <- taxa.data %>% select(`#OTU ID`, ConsensusLineage) %>%
separate(ConsensusLineage, c("Domain", "Phylum", "Class", "Order", "Family", "Genus", "Species"), sep = ";")
sampledata <- t1d_metadata
```

## **Required to have similar row names in OTU and taxonomy table**

```{r include=FALSE}
taxonomy <- column_to_rownames(taxonomy, var = "#OTU ID")
otu.data <- column_to_rownames(otu.data, var = "#OTU ID")
sampledata <- column_to_rownames(sampledata, var = "Sample_ID")
```


## **Change OTU and taxonomy dataframes to matrices**

```{r }
otu.data_matx <- as.matrix(otu.data)
taxonomy_matx <- as.matrix(taxonomy)
```

## **create phyloseq object**

```{r include=FALSE}
otu.data_table <-  otu_table(otu.data_matx, taxa_are_rows = TRUE)
sample_data1 <-  sample_data(sampledata)
taxonomy_table <-  tax_table(taxonomy_matx)

phyloseq.objectA = phyloseq(otu.data_table, taxonomy_table, sample_data1)
```

## **Validity checks**

```{r include=FALSE}
rank_names(phyloseq.objectA)
taxa_names(phyloseq.objectA)
sample_variables(phyloseq.objectA)
```
3. Generate Alpha diversity plots and ordination plots. Examine any observed patterns by delivery mode, gender and disease status.


## **Alpha Diversity**
```{r fig1, fig.height = 6, fig.width = 8, fig.asp = .62}
Fig.A <- plot_richness(phyloseq.objectA,x = "Delivery_Route", color="Case_Control", measures= "Chao1")
Fig.A 
```
Its evident that all there were no cases involved in the sample that had been delivered by cesarian section. impling that all cases involved in the studay had been born by vaginal route. 


```{r fig.height = 6, fig.width = 8, fig.asp = .62}
Fig.B <- plot_richness(phyloseq.objectA,x = "Gender", color="Case_Control", measures= c("Chao1", "fisher"))
Fig.B

```
The distribution shows that, the male and female were equally involved in the study both as cases and controls.


```{r}
Fig.C <- plot_richness(phyloseq.objectA,x = "Age_at_Collection", color="Case_Control", measures= c("fisher", "chao1"))
Fig.C
```
Its evident that, most of the elements involved in the study were bellow 750 days of life and fall within 0 to 150 measure of diversity.


```{r}
Fig.D <- plot_richness(phyloseq.objectA,x = "Age_at_Collection", color="Case_Control", measures= "Chao1")
Fig.D
```
like the Fisher test, the choa1 test of richness also shows that, most of the elements involved in the study were bellow 750 days of life and fall within 0 to 150 measure of diversity.

## **Ordination Plots** 

```{r}

gf <- ordinate(phyloseq.objectA, "PCoA", "bray")%>%
plot_ordination(phyloseq.objectA, .,color = "Gender", title = "Diversity of samples by Gender")+ 
  theme_classic()
gf
```

```{r}
library(phyloseq)
ordinate <- ordinate(phyloseq.objectA, "PCoA", "bray")

gf2 <- plot_ordination(phyloseq.objectA, ordinate,color = "Case_Control", title = "Diversity of samples by Disease Status")+ 
  theme_classic()
gf2
```

```{r}
library(phyloseq)
ordinate1 <-ordinate(phyloseq.objectA, "NMDS", "bray")
gf3 <-plot_ordination(phyloseq.objectA, ordinate,type="taxa", color="Phylum", title = "Diversity of samples by Disease Status")+ 
  theme_classic()
gf3
```



# **4. Perform differential abundance using DEseq2**


## **DESeq: Creating a DESeq object**

```{r}
phyloseq_objectB <- phyloseq(otu_table(otu.data_matx + 1, taxa_are_rows=TRUE), tax_table(taxonomy_matx))
sample_data(phyloseq_objectB) <- sample_data1

casecontrol = phyloseq_to_deseq2(phyloseq_objectB, ~ Case_Control)
```



## **DESeq test**

```{r message=TRUE, warning=FALSE}
casecontrol = DESeq(casecontrol, test="Wald", fitType="parametric")
```

## **Results table**

```{r}
res = results(casecontrol, cooksCutoff = FALSE)
alpha = 0.01
casetab = res[which(res$padj < alpha), ]
casetab = cbind(as(casetab, "data.frame"), as(tax_table(phyloseq_objectB)[rownames(casetab), ], "matrix"))

```


## **plot_theme**
```{r}
theme_set(theme_bw())
scale_fill_discrete <- function(palname = "Set1", ...) {
  scale_fill_brewer(palette = palname, ...)
}

```

## **Phylum order**

```{r}
x = tapply(casetab$log2FoldChange, casetab$Phylum, function(x) max(x))
x = sort(x, TRUE)
casetab$Phylum = factor(as.character(casetab$Phylum), levels=names(x))
```


## **Genus order**

```{r , fig.height = 6, fig.width = 10, fig.asp = .80}
x = tapply(casetab$log2FoldChange, casetab$Genus, function(x) max(x))
x = sort(x, TRUE)
casetab$Genus = factor(as.character(casetab$Genus), levels=names(x))
ggplot(casetab, aes(x=Genus, y=log2FoldChange, color=Phylum)) + geom_point(size=6) +
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))
```




